util/types/json/path_expr.go

```diff
@@ -127,8 +127,12 @@ func ParseJSONPathExpr(pathExpr string) (pe PathExpression, err error) {
 		}
 	}
 
-	pathExprSuffix := pathExpr[dollarIndex+1:]
+	pathExprSuffix := strings.TrimFunc(pathExpr[dollarIndex+1:], isBlank)
 	indices := jsonPathExprLegRe.FindAllStringIndex(pathExprSuffix, -1)
+	if len(indices) == 0 && len(pathExprSuffix) != 0 {
```

Comment:
it's for `select json_extract('{}', '$NoValidPathLegHere')`.
I'll try to add some bad cases such like this.